rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/telemetry/TbMsgAttributesNodeTest.java

```diff
@@ -180,6 +204,97 @@ void givenNotifyDeviceMdValue_whenSaveAndNotify_thenVerifyExpectedArgumentForNot
         assertThat(notifyDevice).isEqualTo(expectedArgumentValue);
     }
 
+    @Test
+    void givenUseTemplateTrueAndNoMetadataKeyInMsg_whenOnMsg_thenSaveAttributes() throws Exception {
+        config.setUseAttributesScopeTemplate(true);
+        config.setScope("${scopeInMetadata}");
+        config.setUpdateAttributesOnlyOnValueChange(false);
+        nodeConfiguration = new TbNodeConfiguration(JacksonUtil.valueToTree(config));
+        node.init(ctx, nodeConfiguration);
+
+        final Map<String, String> mdMap = Map.of("scope", "SERVER_SCOPE");
+        final String data = "{\"TestAttribute_2\": \"humidity\", \"TestAttribute_3\": \"voltage\"}";
+        TbMsg msg = TbMsg.newMsg(TbMsgType.POST_ATTRIBUTES_REQUEST, deviceId, new TbMsgMetaData(mdMap), data);
+
+        node.onMsg(ctx, msg);
+
+        verify(telemetryService).saveAndNotify(eq(ctx.getTenantId()), eq(msg.getOriginator()), eq(AttributeScope.SERVER_SCOPE),
+                anyList(), eq(true), eq(new TelemetryNodeCallback(ctx, msg)));
+        verify(ctx).tellSuccess(eq(msg));
+    }
+
+    @Test
+    void givenUseTemplateTrueAndStrangeDataKey_whenOnMsg_thenSaveAttributes() throws Exception {
+        config.setUseAttributesScopeTemplate(true);
+        config.setScope("$[scopeInData]");
+        config.setUpdateAttributesOnlyOnValueChange(false);
+        nodeConfiguration = new TbNodeConfiguration(JacksonUtil.valueToTree(config));
+        node.init(ctx, nodeConfiguration);
+
+        final String data = "{\"TestAttribute_2\": \"humidity\", \"TestAttribute_3\": \"voltage\", \"scopeInData\": \" SeRvEr_ScOpE \"}";
+        TbMsg msg = TbMsg.newMsg(TbMsgType.POST_ATTRIBUTES_REQUEST, deviceId, TbMsgMetaData.EMPTY, data);
+
+        node.onMsg(ctx, msg);
+
+        verify(ctx).tellSuccess(eq(msg));
```

Comment:
will be nice to have verify of:

verify(telemetryService).saveAndNotify(...) with eq(AttributeScope.SERVER_SCOPE)